### PR TITLE
Pods 7417

### PR DESCRIPTION
**Pull request by cyeniceri**

### CODE REVIEW

File: app/models/enumeration/EventType.scala [hunk 1]
```diff
@@ -83,7 +84,6 @@ object EventType extends Enumerable.Implicits {
       case evType1827 if api1827Events.contains(evType1827) => Some(ApiType.Api1827)
       case evType1829 if api1829Events.contains(evType1829) => Some(ApiType.Api1829)
       case evType1830 if api1830Events.contains(evType1830) => Some(ApiType.Api1830)
-      case _ => None
```
Comment (cyeniceri) [inline app/models/enumeration/EventType.scala:86]: If you remove this, why do we need following else cases ? 
`
  def saveUserAnswers(pstr: String, eventType: EventType, userAnswersJson: JsValue)(implicit ec: ExecutionContext): Future[Unit] = {
    EventType.postApiTypeByEventType(eventType) match {
      // TODO: Have discussion on potential for overwriting in Mongo.
      case Some(apiType) => eventReportCacheRepository.upsert(pstr, apiType, userAnswersJson)
      case _ => Future.failed(new NotFoundException(s"Not Found: ApiType not found for eventType ($eventType)"))
    }
  }
`

and 

`  def getUserAnswers(pstr: String, eventType: EventType)(implicit ec: ExecutionContext): Future[Option[JsObject]] = {
    EventType.postApiTypeByEventType(eventType) match {
      case Some(apiType) =>
        eventReportCacheRepository.getByKeys(Map("pstr" -> pstr, "apiTypes" -> apiType.toString))
          .map(_.map(_.as[JsObject]))
      case _ => Future.failed(new NotFoundException(s"Not Found: ApiType not found for eventType ($eventType)"))
    }
  }`

File: app/models/enumeration/EventType.scala [hunk 2]
```diff
@@ -92,7 +92,7 @@ object EventType extends Enumerable.Implicits {
       case evType1832 if api1832Events.contains(evType1832) => Some(ApiType.Api1832)
       case evType1833 if api1833Events.contains(evType1833) => Some(ApiType.Api1833)
       case evType1831 if api1831Events.contains(evType1831) => Some(ApiType.Api1831)
-      case _ => None
+      case evType1834 if api1834Events.contains(evType1834) => Some(ApiType.Api1834)
```
Comment (cyeniceri) [inline app/models/enumeration/EventType.scala:95]: similarly we don't need this 
In `EventReportConnector.scala`, under `getEvent` method

`case None => throw new BadRequestException(s"No API is configured to handle getting this eventType: $eventType")`

File: app/models/enumeration/EventType.scala (hunk 1 above)
Comment (ngmjohnstone) [reply app/models/enumeration/EventType.scala:86]: Addressed here but with the unchecked annotation - not sure if we'd like to proceed with this approach: https://github.com/hmrc/pension-scheme-event-reporting/pull/36/commits/2c089af273be16623552a522c80200a00f9429a2

File: app/models/enumeration/EventType.scala (hunk 2 above)
Comment (ngmjohnstone) [reply app/models/enumeration/EventType.scala:95]: Addressed here but with the unchecked annotation - not sure if we'd like to proceed with this approach: https://github.com/hmrc/pension-scheme-event-reporting/commit/2c089af273be16623552a522c80200a00f9429a2

Comment (cyeniceri) [reply app/models/enumeration/EventType.scala]: here is simplified with last commit

**Review by ngmjohnstone — APPROVED** [review]
Looks good to me

**Pull request merged**